### ISSUE TITLE
ASM and SIMD instructions for x86 64-bit are always disabled on FreeBSD

### ISSUE THREAD

**Issue by diizzyy** [WayneD/rsync#230]
Unless I'm reading the configure script incorrectly it seems like it assumes that host_cpu is always called x86_64 while on at least FreeBSD it's called amd64.

Running configure script
```
checking whether to enable SIMD optimizations... no
checking if assembler accepts noexecstack... yes
checking whether to enable ASM optimizations... no
```

config.log: 
```host_cpu='amd64'```

FreeBSD 13-STABLE using rsync 3.2.3

**Comment by WayneD** [comment]
I've amended the configure script to allow it to accept amd64 in addition to x86_64, but I'd be cautious with enabling the SIMD & ASM optimizations.  I've had reports from FreeBSD folks (and MacOS folks) that they had to turn SIMD off, so I made both SIMD & ASM default to off on all systems except Linux for the time being.  Feel free to enable and test and see how it goes, but don't make it the default for a lot of people before doing some thorough testing.  I imagine that there must be some weirdness in the build chain on non-Linux systems, but if so, it will take people who use those systems (and who run into issues) to diagnose what is going on.
